ratis-server/src/main/java/org/apache/ratis/server/impl/ReadRequests.java

```diff
@@ -34,7 +37,7 @@ static class HeartbeatAck {
     }
 
     /** Is the heartbeat (for a particular call id) acknowledged? */
-    boolean isAcknowledged() {
+    synchronized boolean isAcknowledged() {
```

Comment:
acknowledged is `volatile` so that we don't have to add `synchronized`.

Reply:
Got it